x/ibc/02-client/client/utils/utils.go

```diff
@@ -14,120 +14,83 @@ import (
 	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
 )
 
-// QueryAllClientStates returns all the light client states. It _does not_ return
-// any merkle proof.
-func QueryAllClientStates(clientCtx client.Context, page, limit int) ([]exported.ClientState, int64, error) {
-	params := types.NewQueryAllClientsParams(page, limit)
-	bz, err := clientCtx.JSONMarshaler.MarshalJSON(params)
-	if err != nil {
-		return nil, 0, fmt.Errorf("failed to marshal query params: %w", err)
+// QueryClientState returns a client state.
+// If prove is true, it performs an ABCI store query in order to retrieve the merkle proof. Otherwise,
+// it uses the gRPC query client.
+func QueryClientState(
+	clientCtx client.Context, clientID string, prove bool,
+) (*types.QueryClientStateResponse, error) {
+	if prove {
+		return QueryClientStateABCI(clientCtx, clientID)
 	}
 
-	route := fmt.Sprintf("custom/%s/%s/%s", "ibc", types.QuerierRoute, types.QueryAllClients)
-	res, height, err := clientCtx.QueryWithData(route, bz)
-	if err != nil {
-		return nil, 0, err
+	queryClient := types.NewQueryClient(clientCtx)
+	req := &types.QueryClientStateRequest{
+		ClientId: clientID,
 	}
 
-	var clients []exported.ClientState
-	err = clientCtx.JSONMarshaler.UnmarshalJSON(res, &clients)
-	if err != nil {
-		return nil, 0, fmt.Errorf("failed to unmarshal light clients: %w", err)
-	}
-	return clients, height, nil
+	return queryClient.ClientState(context.Background(), req)
 }
 
-// QueryClientState queries the store to get the light client state and a merkle
-// proof.
-// TODO: delete
-func QueryClientState(
-	clientCtx client.Context, clientID string, prove bool,
-) (types.StateResponse, error) {
+// QueryClientStateABCI queries the store to get the light client state and a merkle proof.
+func QueryClientStateABCI(
+	clientCtx client.Context, clientID string,
+) (*types.QueryClientStateResponse, error) {
 	req := abci.RequestQuery{
```

Comment:
I think we should abstract this into a function `QueryIBCProof` which takes in the `data` field as an argument. we could put this in `x/ibc/client/query/query.go`. Maybe outside the scope of this pr, but then we just do the proof height + 1 there. 

I can probably do this in a followup, should be pretty easy

Reply:
will leave this for a follow-up #6567